From 6faee50480f61cc6e02eaf35393777bed8615fb4 Mon Sep 17 00:00:00 2001
From: Yong Chen <yong.a.chen@intel.com>
Date: Thu, 2 May 2013 10:28:39 +0800
Subject: [PORT FROM MAIN] Make exception and debug handling compatible with AsmPrinter

BZ: 104927

Add !ANDROID_TARGET_BUILD || ANDROID_ENGINEERING_BUILD when
setting ExceptionsType and SupportsDebugInformation, or else
llvm will encounter invalid frame fatal error.

This patch is for Android llvm only. should not upstream to llvm.org.

Change-Id: I0e5f42670a480c0447c5e89687f26f07d0ce988f
Orig-Change-Id: I72ea6b423197f132a613fce7ae9fec8d23aa4fa1
Signed-off-by: Yong Chen <yong.a.chen@intel.com>
Reviewed-on: http://android.intel.com:8080/105688
Reviewed-by: Ren, Jack <jack.ren@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 lib/Target/X86/MCTargetDesc/X86MCAsmInfo.cpp |   16 +++++++++-------
 1 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/lib/Target/X86/MCTargetDesc/X86MCAsmInfo.cpp b/lib/Target/X86/MCTargetDesc/X86MCAsmInfo.cpp
index 7815ae9..3c606a6 100644
--- a/lib/Target/X86/MCTargetDesc/X86MCAsmInfo.cpp
+++ b/lib/Target/X86/MCTargetDesc/X86MCAsmInfo.cpp
@@ -61,12 +61,13 @@ X86MCAsmInfoDarwin::X86MCAsmInfoDarwin(const Triple &T) {
   CommentString = "##";
   PCSymbol = ".";
 
-  SupportsDebugInformation = true;
   DwarfUsesInlineInfoSection = true;
   UseDataRegionDirectives = MarkedJTDataRegions;
 
-  // Exceptions handling
-  ExceptionsType = ExceptionHandling::DwarfCFI;
+#if !defined(ANDROID_TARGET_BUILD) || defined(ANDROID_ENGINEERING_BUILD)
+  SupportsDebugInformation = true;
+  ExceptionsType = ExceptionHandling::DwarfCFI;  // Exceptions handling
+#endif
 }
 
 X86_64MCAsmInfoDarwin::X86_64MCAsmInfoDarwin(const Triple &Triple)
@@ -98,11 +99,10 @@ X86ELFMCAsmInfo::X86ELFMCAsmInfo(const Triple &T) {
   // Set up DWARF directives
   HasLEB128 = true;  // Target asm supports leb128 directives (little-endian)
 
-  // Debug Information
+#if !defined(ANDROID_TARGET_BUILD) || defined(ANDROID_ENGINEERING_BUILD)
   SupportsDebugInformation = true;
-
-  // Exceptions handling
-  ExceptionsType = ExceptionHandling::DwarfCFI;
+  ExceptionsType = ExceptionHandling::DwarfCFI;  // Exceptions handling
+#endif
 
   // OpenBSD and Bitrig have buggy support for .quad in 32-bit mode, just split
   // into two .words.
@@ -154,5 +154,7 @@ X86MCAsmInfoGNUCOFF::X86MCAsmInfoGNUCOFF(const Triple &Triple) {
   TextAlignFillValue = 0x90;
 
   // Exceptions handling
+#if !defined(ANDROID_TARGET_BUILD) || defined(ANDROID_ENGINEERING_BUILD)
   ExceptionsType = ExceptionHandling::DwarfCFI;
+#endif
 }
-- 
1.7.4.1

